src/theory/datatypes/theory_datatypes.cpp

```diff
@@ -343,6 +343,15 @@ void TheoryDatatypes::preRegisterTerm(TNode n)
       // Get triggered for both equal and dis-equal
       d_state.addEqualityEngineTriggerPredicate(n);
       break;
+    case Kind::MATCH:
+    {
+      Assert (!options().datatypes.datatypesExp);
```

Comment:
Why is this an Assert?

Reply:
If this option is enabled, then we should have eliminated during preprocessing.